src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,333 @@
+   */
+  public static isAnyIPv4AddressInIPv4AddressRange(ips: IPv4Address[], range: IPv4AddressRange): boolean {
+    return ips.filter((ip) => {
+      return this.isInIPv4AddressRange(ip, range);
+    }).length > 0;
+  }
+
+  /**
+   * Converts integer to ip address representation.
+   * @param {number} ipInt: Integer to be converted.
+   * @return {string} IP Address string.
+   */
+  public static getIntegerAsIPv4Address(ipInt: number): string {
+    const components = [];
+    const template = 255;
+    for (let i = 3; i >= 0; i--) {
+      const modTemplate = (template << (i * 8));
+      components.push((modTemplate & ipInt) >>> (i * 8));
+    }
+    return components.join('.');
+  }
+
+  /**
+   * Parses string to int then checks if test number is between them.
+   * @param {string} a first number for comparison.
+   * @param {string} b second number for comparison.
+   * @param {string} test number being compared.
+   * @return {boolean}
+   */
+  public static inIPv4AddressRange(a: string, b: string, test: string): boolean {
+    const aInt = parseInt(a, 10);
+    const bInt = parseInt(b, 10);
+    const testInt = parseInt(test, 10);
+    const min = aInt < bInt ? aInt : bInt;
+    const max = aInt > bInt ? aInt : bInt;
+    return (testInt >= min && testInt <= max);
+  }
+
+  /**
+   * Checks if integer representing port is valid.
+   * @param {number} port port number being checked.
+   * @return {boolean}
+   */
+  public static isValidIPv4AddressPort(port: number): boolean {
+    return (port >= 1 && port <= 65535);
+  }
+
+  /**
+   * Gets the number of IP addresses in a given range. Although unlikely to be
+   * an issue, note that due to the fact that javascript numbers are represented
+   * as signed, two's complement integers, for a network with more than 2^31
+   * addresses this function will return a negative number (May not even be
+   * technically possible). Returns -1 if the parameters are invalid.
+   * @param {IPv4AddressRange} ipRange object representing IP range of the form { start:
+   *          "x.x.x.x", end: "x.x.x.x" }
+   * @return {integer} size of range.
+   */
+  public static getNumberOfIPv4AddressesInIPv4AddressRange(range: IPv4AddressRange): number | null {
+    return this.getIPv4AsInteger(new IPv4Address(range.end)) - this.getIPv4AsInteger(new IPv4Address(range.start)) + 1;
+  }
+
+  /**
+   * Returns an available IP address within one of an array of given ranges,
+   * given an array of the IP addresses that are already allocated. Returns null
+   * if no addresses are unallocated.
+   * @param {IPv4AddressRange[]} ipRanges array of IP ranges to choose from.
+   * @param {string[]} allocatedIpAddresses array of IP addresses which have
+   *          already been allocated.
+   * @return {string} an available IP address.
+   */
+  public static getAvailableIPv4(ipv4AddressRanges: Array<IPv4AddressRange>,
+                                 allocatedIpAddresses: IPv4Address[]): string | null {
```

Comment:
We use Array<> notation for TS array type declarations in this project. I see you using the [] in some other places too, so please update them all.